api/admin/controller.py

```diff
@@ -1934,6 +1945,20 @@ def collections(self):
                         _("The collection configuration is missing a required setting: %(setting)s",
                           setting=setting.get("label")))
                 collection.external_account_id = value
+            elif key == 'mirror_integration_id':
+                value = flask.request.form.get(key)
+                if value == self.NO_MIRROR_INTEGRATION:
+                    integration_id = None
+                else:
+                    integration = get_one(
+                        self._db, ExternalIntegration, id=value
+                    )
+                    if not integration:
+                        return MISSING_SERVICE
+                    if integration.goal != ExternalIntegration.STORAGE_GOAL:
+                        return INTEGRATION_GOAL_CONFLICT
```

Comment:
I usually rollback before returning a problem detail so that none of the changes get saved when there's an error.